client/my-sites/site-settings/form-general.jsx

```diff
@@ -113,57 +111,129 @@ const FormGeneral = React.createClass( {
 			amp_is_supported: false,
 			amp_is_enabled: false,
 		} );
-	},
+		this.props.clearDirtyFields();
+		this.props.updateFields( this.getFormSettings( this.props.settings ) );
+	}
 
-	onTimezoneSelect( timezone ) {
-		this.setState( { timezone_string: timezone } );
-	},
+	componentWillReceiveProps( nextProps ) {
+		this._showWarning( nextProps.site );
+
+		if ( nextProps.settings !== this.props.settings ) {
+			let newState = this.getFormSettings( nextProps.settings );
+			//If we have any fields that the user has updated,
+			//do not wipe out those fields from the poll update.
+			newState = omit( newState, nextProps.dirtyFields );
+			return nextProps.updateFields( newState );
+		}
+	}
 
 	onRecordEvent( eventAction ) {
 		return this.recordEvent.bind( this, eventAction );
-	},
+	}
 
 	onRecordEventOnce( key, eventAction ) {
 		return this.recordEventOnce.bind( this, key, eventAction );
-	},
+	}
+
+	recordEvent( eventAction ) {
```

Comment:
Since we're replacing `formBase.recordEvent`, I think we should do so using newer `state/analytics` actions. Might be an opportunity to introduce a new `recordEventOnce` action creator (separate PR?).